crypto/fipsmodule/ec/oct.c

```diff
@@ -142,7 +152,9 @@ static int ec_GFp_simple_oct2point(const EC_GROUP *group, EC_POINT *point,
   }
 
   point_conversion_form_t form = buf[0];
-  if (form == POINT_CONVERSION_UNCOMPRESSED) {
+  const int y_bit = form & 1;
+  form = form & ~1u;
```

Comment:
please explain the magic & ~1u?

Reply:
`~1u` takes an unsigned `1` and inverts all bits, which leaves behind `...1110`.
`form & ~1u` then performs an AND operation on `form` and `~1u`, which clears the far right bit of `form`. `form` is then adjusted to the nearest even value smaller than or equal to its original value..

Reply:
So that code just sets the least significant bit of `form` to 0 which:
- does nothing if `form` is even,
- subtracts 1 if `form` is odd.

We do that because the hybrid form can be represented by either 6 or 7, but we internally encode it with `POINT_CONVERSION_HYBRID` which is equal to 6.

Anyway, consider adding a properly named helper function that does this for you and also shortly explains what I wrote above. For example:
```
int is_point_conversion_form_hybrid(form) { return POINT_CONVERSION_HYBRID == (form & ~1u); }

Reply:
No problem will look into doing something for that to share the logic.